backend/origuide/fold/test_fold_producer.py

```diff
@@ -0,0 +1,45 @@
+import unittest
+
+from .fold_encoder import LogFoldEncoder
+
+
+class TestLogFoldEncoder(unittest.TestCase):
+    def test_next_step_resets_encoder(self):
+        encoder = LogFoldEncoder(4, 2, 1)
```

Comment:
Personal preference:
I'd use named parameters for primitive types. `LogFoldEncoder(4, 2, 1)` is very cryptic and requires reader to navigate to the implementation.

`LogFoldEncoder(frame_drop_rate=4,
                 frame_drop_multiplier=2,
                 frame_drop_change_iter=1)` looks more descriptive

Reply:
I agree. I'll leave it like that in tests but change in main.